PusherServer/RequestResult.cs

```diff
@@ -10,6 +11,11 @@ public abstract class RequestResult: IRequestResult
 
         public RequestResult(IRestResponse response)
         {
+            if (response == null)
+            {
+                throw new ArgumentNullException("response");
+            }
+
```

Comment:
What does it bring above having an exception when response.Content is accessed ?